paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/samples/ui/wallet/SavedPaymentMethodsViewState.kt

```diff
@@ -1,5 +1,10 @@
 package com.stripe.android.paymentsheet.example.samples.ui.wallet
 
+import com.stripe.android.ExperimentalSavedPaymentMethodsApi
+import com.stripe.android.paymentsheet.repositories.CustomerEphemeralKey
+
+@OptIn(ExperimentalSavedPaymentMethodsApi::class)
 data class SavedPaymentMethodsViewState(
```

Comment:
I find this slightly harder to follow than a sealed class alternative.

What if we did something like this instead?

```
sealed class SavedPaymentMethodsViewState {
    object Loading : SavedPaymentMethodsViewState()

    class FailedToLoad(val message: String) : SavedPaymentMethodsViewState()

    class Data(val customerEphemeralKey: CustomerEphemeralKey) : SavedPaymentMethodsViewState()
}
```

Reply:
I like it!